teos/src/responder.rs

```diff
@@ -219,28 +224,38 @@ impl Responder {
     /// Returns the set of completed trackers or [None] if none were completed.
     fn check_confirmations(&self, txids: HashSet<Txid>, current_height: u32) -> Option<Vec<UUID>> {
         let mut completed_trackers = Vec::new();
+        let mut reorged_trackers = self.reorged_trackers.lock().unwrap();
         let dbm = self.dbm.lock().unwrap();
 
         for (uuid, (penalty_txid, status)) in dbm.load_penalties_summaries() {
-            if let ConfirmationStatus::ConfirmedIn(h) = status {
-                let confirmations = current_height - h;
-                if confirmations == constants::IRREVOCABLY_RESOLVED {
-                    // Tracker is deep enough in the chain, it can be deleted
-                    completed_trackers.push(uuid);
-                } else {
-                    log::info!("{uuid} received a confirmation (count={confirmations})");
-                }
-            } else if txids.contains(&penalty_txid) {
+            if txids.contains(&penalty_txid) {
                 // First confirmation was received
                 dbm.update_tracker_status(uuid, &ConfirmationStatus::ConfirmedIn(current_height))
                     .unwrap();
-            } else if let ConfirmationStatus::InMempoolSince(h) = status {
-                // Log all transactions that have missed confirmations
-                log::info!(
-                    "Transaction missed a confirmation: {} (missed conf count: {})",
-                    penalty_txid,
-                    current_height - h
-                );
+                // Remove that uuid from reorged trackers if it was confirmed.
+                reorged_trackers.remove(&uuid);
+            } else {
+                match status {
+                    // We will end up here with h >= current_height if this tracker was confirmed on another fork (reorg).
+                    ConfirmationStatus::ConfirmedIn(h) if current_height > h => {
+                        let confirmations = current_height - h;
+                        if confirmations == constants::IRREVOCABLY_RESOLVED {
+                            // Tracker is deep enough in the chain, it can be deleted
+                            completed_trackers.push(uuid);
+                        } else {
+                            log::info!("{uuid} received a confirmation (count={confirmations})");
+                        }
+                    }
+                    ConfirmationStatus::InMempoolSince(h) if current_height > h => {
```

Comment:
I don't think this is right.

We should never get here with an `h >= current_height` in any of the cases. This is because during a reorg blocks are disconnected before they start getting connected again (so we go down, then up). During disconnections, every block that gets disconnected flags anything on its `h` as `ReorgedOut`. 

Let `h_n` be the last block height disconnected during a reorg, the first connected height will have `current_height = h_n+1`, and all trackers that we're keeping track of between `h_n` and `h_old_tip` will be marked as `ReorgedOut`. Hence, `h` must be strictly smaller than `current_height`.